File: src/debug/entry/javascript/problems/133.js
function runUserScript(userFunm, params, paramTypes) {
    if (params.length !== paramTypes.length) {
        onParameterError();
    }

    const parsedParams = params.map((param, index) => {
        const type = paramTypes[index];
        return parseParameter(index, type, param);
    });
    userFunm.apply(null, parsedParams);
}

function Node(val, neighbors) {
    this.val = val;
    this.neighbors = neighbors;
}

/**
 * @param {number[][]} param
 */
function parseNode(nums) {
    const arr = [];
    nums.map((n, i) => {
        arr.push(new Node(i + 1, []));
    });
    nums.map((nei, i) => {
        nei.map(k => {
            arr[i].neighbors.push(arr[k - 1]);
        });
    });
    return arr[0];
}

function parseSpecialParameter(index, type, param) {
    switch (type) {
        case "Node":
            return parseNode(param);
    }
    return null;
}

File: src/debug/entry/python3/problems/133.py
def runUserScript(func, params, paramTypes):
    if (len(params) != len(paramTypes)):
        onParameterError()

    newParams = []
    for i, val in enumerate(params):
        newParams.append(parseParameter(i, paramTypes[i], val))
    func(*newParams)


class Node:
    def __init__(self, val, neighbors):
        self.val = val
        self.neighbors = neighbors


def parseNode(param, nodeMap):
    arr = []
    for i, val in enumerate(param):
        arr.append(Node(i + 1, []))

    for i, val in enumerate(param):
        for j, k in enumerate(val):
            arr[i].neighbors.append(arr[k - 1])

    return arr[0]


def parseSpecialParameter(index, paramType, param):
    if paramType == "Node":
        return parseNode(param, None)
    return None
